docs/plugin/interfaces/interfaces.md

```diff
@@ -196,44 +202,61 @@ For a more detailed view, visit [Messenger.h](https://github.com/WebPlatformForE
 
-	virtual Core::hresult Create(uint32_t& id /* @out */) = 0;
-	virtual Core::hresult Configure(const string& config) = 0;
-}
-```
-An example of possible calls to the interface: the lookup method is used to convert the id to the object.
+If you are done using the object you can call the [relinquish](https://github.com/rdkcentral/ThunderNanoServices/blob/development/generator-showcase-exmaple-plugin/examples/GeneratorShowcase/docs/GeneratorShowcasePlugin.md#method_relinquish) method specifying the object ID of the object you want to destroy.
+
+The IDevice also has an event, see [here](https://github.com/rdkcentral/ThunderNanoServices/blob/bbbfdb7027a3afb74c4fb1c90981dfc51de37eb4/examples/GeneratorShowcase/interfaces/ISimpleInstanceObjects.h#L43).
+
+To register for an event for a specific Device, so object ID in the JSON-RPC world you can see [here](https://github.com/rdkcentral/ThunderNanoServices/blob/development/generator-showcase-exmaple-plugin/examples/GeneratorShowcase/docs/GeneratorShowcasePlugin.md#head_Notifications) how that is done, again specify the object ID with the # delimiter in the designator for the register method. As you can see when the event is sent the object ID is also in the event designator (convenient in case you registered for notifications of multiple object ID's)
+
+[Here](https://github.com/rdkcentral/ThunderNanoServices/blob/bbbfdb7027a3afb74c4fb1c90981dfc51de37eb4/examples/GeneratorShowcase/GeneratorShowcase.h#L673) you can see how to call the generated Closed method to do cleanup in case the channel closed without all devices being Relinquished. As mentioned it is also possible to pass a callback to the Closed method to get notified on all devices being released here (not demonstrated).
+
+<hr/>
+
+#### Asynchronous Functions
+
+When an action triggered by a method on a COM-RPC interface which takes some time to complete this method will be made asynchronous, meaning it will return when the action is started and it will expect a callback interface to be passed as input parameter to the method so a method on the callback interface will be called when the action that was started is finished or failed (it it mandatory that the started action always results in a method to be called so ti is clear when the action is over).
+
+This is also supported on the JSON-RPC interface. Of course it is not possible to pass a callback so here the end of the action will be indicated by an event send on the same channel the action was started on and an async event ID will be used to connect the function that started the action to the event. Use the @async tag to indicate the method should be async on the JSON-RPC interface (it is of course expected to have one input callback interface parameter and the interface for the callback should contain one callback method only).
```

Comment:
Is it worth mentioning that you don't need to explicitly register for this event with "register" JSON-RPC call?

Reply:
Yes that is a good idea, indeed although logical perhaps it could be unclear so let me add that